File: examples_android/app/src/main/java/com/confirmit/testsdkapp/viewmodels/OverviewViewModel.kt
package com.confirmit.testsdkapp.viewmodels

import com.confirmit.mobilesdk.SurveySDK
import com.confirmit.testsdkapp.viewmodels.models.SurveyModel

class OverviewViewModel(serverId: String, surveyId: String) {
    var survey: SurveyModel = SurveyModel(serverId, surveyId)
        private set

    fun updateSelectedLanguage(langId: Int) {
        survey.survey.languages.first { it.id == langId }.let {
            survey.selectedLanguage = it
        }
    }

    fun removeSurvey(): Boolean {
        return try {
            SurveySDK.deleteSurvey(survey.survey.serverId, survey.survey.surveyId)
            true
        } catch (_: Exception) {
            false
        }
    }
}

File: examples_ios/MyApp/Sources/ViewModels/OverviewViewModel.swift
import Foundation
import ConfirmitMobileSDK

class OverviewViewModel {
    private(set) var survey: SurveyModel!
    
    init(serverId: String, surveyId: String) throws {
        survey = try SurveyModel(serverId: serverId, surveyId: surveyId)
    }
    
    func updateSelectedLanguage(langId: Int) {
        if let language = self.survey.survey.languages.first(where: { $0.id == langId }) {
            survey.selectedLanguage = language
        }
    }
    
    func removeSurvey() -> Bool {
        do {
            try SurveySDK.deleteSurvey(serverId: survey.survey.serverId, surveyId: survey.survey.surveyId)
            return true
        } catch {
            return false
        }
    }
}
